src/alexandria3k/data_sources/uspto.py

```diff
@@ -18,71 +18,621 @@
+        # pylint: disable-next=consider-using-with
+        for path in self.file_path:
+            self.unique_patent_xml_files = self.get_zip_contents(path)
+            # Make the number constant
+            self.filename = path[BASENAME_POINT:]
+            self.container_id = -1
+            self.zip_path = path
+            for patent in self.unique_patent_xml_files:
+                if patent is None:
+                    continue
+
+                self.container_id += 1
+                yield self.container_id
+
+    def get_current_zip_path_by_id(self, zip_file_id):
+        """Return the path of the current Zip file, using id."""
+        self.zip_path = self.file_path[zip_file_id]
+        return self.zip_path
+
+    def get_current_zip_path(self):
+        """Return the path of the current Zip file, updates from generator."""
+        return self.zip_path
+
+    def get_container_id(self):
+        """Return the container id of the XML chunk."""
+        return self.container_id
+
+    def length_of_zip_files(self):
+        """Return the length of the array of data files"""
+        return len(self.file_path)
+
+    def get_container_iterator(self):
+        """Return an iterator over the int identifiers of all chunks XML data files"""
+        return self.zip_generator()
+
+    def get_container_name(self, fid):
+        """Return the name of the file."""
+        return self.filename
+
+
+class VTSource:
+    """Virtual table data source.  This gets registered with the apsw
+    Connection through createmodule in order to instantiate the virtual
+    tables."""
+
+    def __init__(self, data_directory, sample):
+        self.data_files = ZipFiles(data_directory, sample)
+        self.table_dict = {t.get_name(): t for t in tables}
+        self.sample = sample
+
+    def Create(self, _db, _module_name, _db_name, table_name):
+        """Create the specified virtual table
+        Return the tuple required by the apsw.Source.Create method:
+        the table's schema and the virtual table class."""
+        table = self.table_dict[table_name]
+        return table.table_schema(), StreamingCachedContainerTable(
+            table,
+            self.table_dict,
+            self.data_files,
+            self.sample,
+        )
+
+    Connect = Create
+
+    def get_container_iterator(self):
+        """Return an iterator over the data files identifiers"""
+        return self.data_files.get_container_iterator()
+
+    def get_container_name(self, fid):
+        """Return the name of the file corresponding to the specified fid"""
+        return self.data_files.get_container_name(fid)
+
+
+class ChunkCursor(ItemsCursor):
+    """A virtual table cursor over patents data.
+    If it is used through data_source partitioned data access
+    within the context of a ZipFiles iterator,
+    it shall return the single element of the ZipFiles iterator.
+    Otherwise it shall iterate over all elements."""
```

Comment:
Layering issue in the comment.  This lower-level class should not have assumptions on how it is used by upper layers. So write the description in a bit more abstract manner.  It is OK to mention special cases as optimizations.